blockchain/validate.go

```diff
@@ -556,16 +565,18 @@ func CheckBlockSanity(block *btcutil.Block, powLimit *big.Int, timeSource Median
 	return checkBlockSanity(block, powLimit, timeSource, BFNone)
 }
 
-// checkSerializedHeight checks if the signature script in the passed
-// transaction starts with the serialized block height of wantHeight.
-func checkSerializedHeight(coinbaseTx *btcutil.Tx, wantHeight int64) error {
+// ExtractCoinbaseHeight attempts to extract the height of the block
+// from the scriptSig of a coinbase transaction. Coinbase heights
+// are only present in blocks of version 2 or later. This was added as part of
+// BIP0034.
+func ExtractCoinbaseHeight(coinbaseTx *btcutil.Tx) (uint64, error) {
```

Comment:
This should probably return a signed int (I'd argue for int32, but that ship has sailed).